lib/writer.h

```diff
@@ -0,0 +1,21 @@
+#ifndef WRITER_H
+#define WRITER_H
+
+#define MAX_VALUE_LEN 512
+
+#define EXEC_MODE_RECORD 0
+#define EXEC_MODE_REPLAY 1
+
+struct logline {
+    int id;
+    int event_type;
+    char value[MAX_VALUE_LEN];
+};
+
+extern int record(int event_t, char *val);
```

Comment:
Should we just write a logline instead of the component parts? Then the logger can overwrite the values that it itself is supposed to generate, like the timestamp.